### PR TITLE
chore: read DPI scaling from X resources

### PR DESCRIPTION
**Pull request by ramezgerges**
GitHub Issue (If applicable): closes https://github.com/unoplatform/uno-private/issues/478

## PR Type

What kind of change does this PR introduce?

## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] Docs have been added/updated which fit [documentation template](https://github.com/unoplatform/uno/blob/master/doc/.feature-template.md) (for bug fixes / features)
- [ ] [Unit Tests and/or UI Tests](https://github.com/unoplatform/uno/blob/master/doc/articles/uno-development/working-with-the-samples-apps.md) for the changes have been added (for bug fixes / features) (if applicable)
- [ ] Validated PR `Screenshots Compare Test Run` results.
- [ ] Contains **NO** breaking changes
- [ ] Associated with an issue (GitHub or internal) and uses the [automatic close keywords](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue).
- [ ] Commits must be following the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/#summary) specification.

## Other information

Internal Issue (If applicable):

### CODE REVIEW

**Comment by ramezgerges** [comment]
needs to be rebased after https://github.com/unoplatform/uno/pull/17171 merges

File: src/Uno.UI.Runtime.Skia.X11/X11DisplayInformationExtension.cs [hunk 1]
```diff
@@ -232,6 +288,32 @@ private static float FloorScale(double rawDpi)
...
+					if (Marshal.PtrToStringAnsi(value.addr, (int)value.size) is { } str && int.TryParse(str, out var result))
+					{
+						scaling = result / DisplayInformation.BaseDpi;
+						return true;
+					}
+				}
+			}
+			scaling = null;
+			return false;
+		}
```
Comment (jeromelaban) [inline src/Uno.UI.Runtime.Skia.X11/X11DisplayInformationExtension.cs:315]: Let's exact another method called TryGetXResource with Xft.dpi as a parameter.

**Comment by jeromelaban** [comment]
@mergifyio backport release/stable/5.3

File: src/Uno.UI.Runtime.Skia.X11/X11XamlRootHost.x11events.cs [hunk 2]
```diff
@@ -144,97 +145,139 @@ static IEnumerable<XEvent> GetEvents(IntPtr display)
...
+					{
+						case XEventName.PropertyNotify:
+							using (X11Helper.XLock(x11Window.Display))
+							{
+								if (@event.PropertyEvent.atom == X11Helper.GetAtom(x11Window.Display, X11Helper.RESOURCE_MANAGER))
+								{
+									if (this.Log().IsEnabled(LogLevel.Information))
+									{
+										this.Log().Info($"X resources changed. Updating DPI scaling.");
+									}
```
Comment (jeromelaban) [inline src/Uno.UI.Runtime.Skia.X11/X11XamlRootHost.x11events.cs:165]: Let's move this to debug instead.

**Review by jeromelaban — APPROVED** [review]

File: src/Uno.UI.Runtime.Skia.X11/X11DisplayInformationExtension.cs [hunk 3]
```diff
@@ -68,13 +68,68 @@
...
+// that the above copyright notice appear in all copies and that both that
+// copyright notice and this permission notice appear in supporting
+// documentation, and that the name of Digital Equipment Corporation not be
+// used in advertising or publicity pertaining to distribution of the software
+// without specific, written prior permission.
+//
+// ----------------------------------------------------------------
+//
+// Copyright 1991, Digital Equipment Corporation.
+// Copyright 1991, 1994, 1998  The Open Group
```
Comment (carldebilly) [inline src/Uno.UI.Runtime.Skia.X11/X11DisplayInformationExtension.cs:98]: @jeromelaban Should we have any special handling for such external copyrights?

Comment (jeromelaban) [reply src/Uno.UI.Runtime.Skia.X11/X11DisplayInformationExtension.cs:98]: This is preserving the intent of the original copyright.

**Pull request merged**